src/main/java/uk/gov/companieshouse/items/orders/api/controller/CertificateItemsController.java

```diff
@@ -66,10 +85,15 @@ public CertificateItemsController(final CreateItemRequestValidator createItemReq
     public ResponseEntity<Object> createCertificateItem(final @Valid @RequestBody CertificateItemDTO certificateItemDTO,
                                                         HttpServletRequest request,
                                                         final @RequestHeader(REQUEST_ID_HEADER_NAME) String requestId) {
-        trace("ENTERING createCertificateItem(" + certificateItemDTO + ")", requestId);
+        Map<String, Object> logMap = new HashMap<>();
+        logMap.put(REQUEST_ID_LOG_KEY, requestId);
+        LOGGER.infoRequest(request, "create certficate item request", logMap);
 
         final List<String> errors = createItemRequestValidator.getValidationErrors(certificateItemDTO);
         if (!errors.isEmpty()) {
+            logMap.put(ERRORS_LOG_KEY, errors);
```

Comment:
Are we intentionally relying upon the accumulation of map key value pairs as we read down through the code? This raises questions like this in the code reader's head. What scope is there for refactoring some of this behaviour into functions to reduce code duplication and make it more obvious to the code reader what is intended to happen at any point?

Reply:
I'll extract some of the common calls e.g. creating the map with a request id value in it, and adding errors with a bad request. Otherwise there isn't much similarity with the data being stored (not sure of the value of extracting recording of a status to a method) so the accumulative data helps to provide an easy data trace in the logs